Feature order content

Shirosaya25 opened the pull request:
Don't really know how much should be added at the same time, so trying a merge.

- Added the feature: Ability to rearrange content within a module.
- This required an update to the backend to include a new column in an existing table and an update to the models affected as well.
- Added Angular Material to the project and switched out the base card class for mat-card
- Added methods to persist content priority in frontend.
- Fixed / refactored tests that were failing or now irrelevant

- Notes: This will probably break the current application because the backend needs to be updated as well in order for it to work properly. I believe that displaying modules should still work, but anything other than that that directly modifies or adds content to a module will most likely be broken until the database is updated

Shirosaya25 committed:
test

Shirosaya25 committed:
First commit order content

Shirosaya25 committed:
personal branch commit

Shirosaya25 committed:
Refactored frontend to achieve feature completion

Shirosaya25 committed:
Priority

Shirosaya25 committed:
Comments and fixed recursive things

Shirosaya25 committed:
Styling

Shirosaya25 committed:
UI and submodule deletion

Shirosaya25 committed:
Recommented Code, Switched to Angular Material, Broke alot of stylnig on frontend as a result

Shirosaya25 committed:
Merge pull request #151 from revaturelabs/feature-order-content-Michael-Zhang

Ordering content completed

Shirosaya25 committed:
Working frontend to send to Kyle's team: Implemented backend functionality for links

Shirosaya25 committed:
Merge pull request #152 from revaturelabs/feature-order-content-Michael-Zhang

Somewhat working frontend to base off of

Shirosaya25 committed:
Fixed Tests / Refactored them: fixed opening tabs

Shirosaya25 committed:
Merge pull request #153 from revaturelabs/feature-order-content-Michael-Zhang

Fixed Tests / Refactored them: fixed opening tabs

carolynrehm commented on src/main.ts:
```diff
@@ -1,3 +1,4 @@
+import 'hammerjs';
```
Why is this included in the main.ts file?

Shirosaya25 replied:
Loaded by Angular Material since that's one of their dependencies

carolynrehm commented on src/index.html:
```diff
@@ -22,6 +22,8 @@
     <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" />
     <script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/4.0.0-beta.2/js/bootstrap.bundle.min.js"></script>
 
+    <link href="https://fonts.googleapis.com/css?family=Roboto:300,400,500" rel="stylesheet">
```
These should be included with npm and included in the styles array rather than in the index.html

Shirosaya25 replied:
I think that was auto generated with including Angular Material. Do you have any idea on how to include it with npm?

carolynrehm replied:
Okay, if this was generated we don't need to worry about it

carolynrehm commented on src/app/components/content-creator-page/content-creator-page.component.ts:
```diff
@@ -166,6 +166,8 @@ export class ContentCreatorPageComponent implements OnInit {
          this.description, this.url,
          this.getLinksFromSubjects(this.selectedSubjects));
 
+      console.log(content);
```
remove unnecessary comments and console logs

Shirosaya25 replied:
Should have removed all of the console.logs from the ts files in latest commit

Shirosaya25 committed:
Removed console.logs

carolynrehm commented on src/app/components/module-index-page/module-index-page.component.ts:
```diff
@@ -1,193 +1,389 @@
-                  this.toastr.error('Response was null');
-               }
+    /** Map of Visibility status of each Module */
+    contentVisible: Map<Module, boolean> = new Map<Module, boolean>();
```
Is a map necessary here? Can multiple Modules be visible at once?

Shirosaya25 replied:
Made the map into a variable

carolynrehm replied:
What then is the difference between the active module and the contentVisible and selModule? They seem to all be referencing the same thing

carolynrehm replied:
+contentActive

Shirosaya25 replied:
Apparently nothing. I squashed contentVisible and selModule and changed the references to them to activeModule

Shirosaya25 replied:
contentActive is the mapping of what piece of content is considered active in each module. That should be a map since it tracks which piece is active even if you close that module

carolynrehm commented on src/app/components/module-index-page/module-index-page.component.ts:
```diff
@@ -1,193 +1,389 @@
+     */
+    removeModule() {
+
+        const selMethod = (document.getElementById('Seldelmethod') as HTMLInputElement).value;
```
Why use the document object here rather than databinding?

Shirosaya25 replied:
Artifact of last batch that worked on it. Removed and attached ngModel in the HTML

Shirosaya25 committed:
TS

Shirosaya25 committed:
Removed raw document call; Made the map into a module variable

Shirosaya25 committed:
Fixed the Tests

Shirosaya25 committed:
Fixed Resolve Requests Styling

Shirosaya25 committed:
Removed contentVisible and selModule; Changed refs to activeModule

Shirosaya25 committed:
Fixed tests.. again

carolynrehm commented on src/app/models/Module.ts:
```diff
@@ -1,6 +1,8 @@
 import { Link } from './Link';
 import { ReqLink } from './ReqLink';
 
+// TODO
```
please remove this

Shirosaya25 replied:
Removed

Shirosaya25 committed:
Removed TODO

carolynrehm commented on src/app/pipes/content-filter.pipe.ts:
```diff
@@ -0,0 +1,24 @@
+    name: 'contentFilter'
+})
+
+export class ContentFilterPipe implements PipeTransform {
```
Where is this filter used?

Shirosaya25 replied:
Feature I was going to implement but deemed to complicated. Deleted in the next commit

Shirosaya25 committed:
Removed ModuleIndex ContentFilter

carolynrehm reviewed (changes requested):
Is there a reason why a pipe needs to be used to order content? Angular generally [advises against it](https://angular.io/guide/pipes#appendix-no-filterpipe-or-orderbypipe).  It seems like we only need an initial ordering, so we could just provide a sorted array of lists to begin with.

Furthermore, sorting and filtering operations like these, ideally, should be performed on the back end. If these cannot be changed, it might be better to rename/reorganize the service. It's hard to know what "UtilService" would provide

Shirosaya25 commented:
I tried removing the pipe for ordering content and everything seems to work correctly.

I also renamed util to sortSearch and replaced the refs where needed

Shirosaya25 committed:
Removed redundant pipes, renamed util service

Shirosaya25 committed:
Fixed Tests

carolynrehm commented on src/app/components/module-index-page/module-index-page.component.html:
```diff
@@ -93,11 +93,11 @@
                                             <em style="width:20px;" class="fa" [ngClass]="{'fa-file-code-o': link.content.format === 'Code', 'fa-file-text-o': link.content.format === 'Document', 'fa-file-powerpoint-o': link.content.format === 'Powerpoint'}"></em>
 
-                                            &nbsp; {{link.content.title}}
+                                            &nbsp;{{link.priority}} - {{link.content.title}}
```
Why add this explicitly?

Shirosaya25 replied:
forgot to remove it when testing to make sure it functioned without the pipe

Shirosaya25 committed:
stopped showing priority

carolynrehm reviewed (approved):
Looks good! I would wait until the back end branch is ready to merge though

carolynrehm commented:
This pull requests should not include content ordering. Merge that code into dev or remove the unrelated implementation from this pull request

carolynrehm commented:
@Shirosaya25 a commit has been merged into dev, pull these updates into this branch and it will be merged

Shirosaya25 committed:
Merge branch 'dev' of https://github.com/revaturelabs/cms-angular into feature-order-content

Shirosaya25 commented:
@carolynrehm I pulled the changes from dev. Just waiting for Travis to finish
